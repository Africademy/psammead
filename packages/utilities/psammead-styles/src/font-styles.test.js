import {
  getSansRegular,
  getSansItalic,
  getSansBold,
  getSansBoldItalic,
  getSerifMedium,
  getSerifMediumItalic,
} from './font-styles';

describe('fonts', () => {
  describe('getSansRegular', () => {
    it('should return regular sans font css', () => {
      expect(getSansRegular('igbo')).toMatchSnapshot();
    });

    it('should fail gracefully given an invalid service name', () => {
      expect(getSansRegular('default')).toBeUndefined();
    });
  });

  describe('getSansItalic', () => {
    it('should return regular sans italic font css', () => {
      expect(getSansItalic('yoruba')).toMatchSnapshot();
    });
  });

  describe('getSansBold', () => {
    it('should return bold sans font css', () => {
      expect(getSansBold('pidgin')).toMatchSnapshot();
    });
  });

  describe('getSansBoldItalic', () => {
    it('should return bold italic font css', () => {
      expect(getSansBoldItalic('igbo')).toMatchSnapshot();
    });
  });

  describe('getSerifMedium', () => {
    it('should return medium serif font css', () => {
      expect(getSerifMedium('news')).toMatchSnapshot();
    });
    it('should return regular sans bold font css for sevices without serif medium', () => {
      expect(getSerifMedium('igbo')).toMatchSnapshot();
    });
  });

  describe('getSerifMediumItalic', () => {
    it('should return serif medium italic font css', () => {
      expect(getSerifMedium('news')).toMatchSnapshot();
    });
    it('should return sans bold italic font css for sevice without serif medium italic', () => {
      expect(getSerifMediumItalic('yoruba')).toMatchSnapshot();
    });
  });
});
