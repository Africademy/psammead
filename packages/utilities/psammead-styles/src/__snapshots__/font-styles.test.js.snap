// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fonts getSansBold should return bold sans font css 1`] = `
"
    font-family: Helmet, Freesans, Helvetica, Arial, sans serif;
    font-weight: 500;
    font-style: normal;
  "
`;

exports[`fonts getSansBoldItalic should return bold italic font css 1`] = `
"
    font-family: Helmet, Freesans, Helvetica, Arial, sans serif;
    font-weight: 500;
    font-style: italic;
  "
`;

exports[`fonts getSansItalic should return regular sans italic font css 1`] = `
"
    font-family: Helmet, Freesans, Helvetica, Arial, sans serif;
    font-weight: 400;
    font-style: italic;
  "
`;

exports[`fonts getSansRegular should return regular sans font css 1`] = `
"
    font-family: Helmet, Freesans, Helvetica, Arial, sans serif;
    font-weight: 400;
    font-style: normal;
  "
`;

exports[`fonts getSerifMedium should return medium serif font css 1`] = `
"
   font-family: ReithSerif, Helvetica, Arial, sans-serif; 
   font-style: normal;
   font-weight: 500;
  "
`;

exports[`fonts getSerifMedium should return regular sans bold font css for sevices without serif medium 1`] = `
"
    font-family: Helmet, Freesans, Helvetica, Arial, sans serif;
    font-weight: 500;
    font-style: normal;
  "
`;

exports[`fonts getSerifMediumItalic should return sans bold italic font css for sevice without serif medium italic 1`] = `
"
    font-family: Helmet, Freesans, Helvetica, Arial, sans serif;
    font-weight: 500;
    font-style: italic;
  "
`;

exports[`fonts getSerifMediumItalic should return serif medium italic font css 1`] = `
"
   font-family: ReithSerif, Helvetica, Arial, sans-serif; 
   font-style: normal;
   font-weight: 500;
  "
`;
